kebechet/payload_parser.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python3
+# Kebechet
+# Copyright(C) 2020 Sai Sankar Gochhayat
+#
+# This program is free software: you can redistribute it and / or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Provides abstraction of payload parsing functions."""
+
+from .enums import ServiceType
+import logging
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class PayloadParser():
+    """Allow Kebechet to parse webhook payload of different services."""
+
+    _GITHUB = "api.github.com"
+    _GITLAB = "gitlab.com"
+
+    _IGNORED_GITHUB_EVENTS = ['installation', 'integration_installation']
+
+    def __init__(self, payload: dict):
+        """Initialize the parameters we require from the services."""
+        self.service_type = None
+        self.url = None
+        self.event = None
+
+        # For github webhooks
+        if 'event' in payload:
+            if payload['event'] not in self._IGNORED_GITHUB_EVENTS:
+                github_payload = payload['payload']
+                if self._GITHUB in github_payload['sender']['url']:
```

Comment:
What if I create a repo named "api.github.com"? :)

https://docs.python.org/3.7/library/urllib.parse.html?highlight=urlparse#urllib.parse.urlparse

See `netloc` attribute.

The same for gitlab case.

Reply:
Added it :smiley: